[BOLT][AArch64] Handle PAuth call instructions in isIndirectCall

atrosinenko opened the pull request:
Handle `BLRA*` opcodes in AArch64MCPlusBuilder::isIndirectCall, update getRegUsedAsCallDest accordingly.

atrosinenko committed:
[BOLT][AArch64] Handle PAuth call instructions in isIndirectCall

Handle `BLRA*` opcodes in AArch64MCPlusBuilder::isIndirectCall, update
getRegUsedAsCallDest accordingly.

atrosinenko commented:
This PR is intended to be a follow-up for #131899.

jacobbramley reviewed (approved):
It looks sensible to me!

atrosinenko commented:
@jacobbramley Do you suggest merging this first and then updating #131899? My initial idea was to first merge #131899 after addressing the remaining comments, then update this PR (by changing `getRegUsedAsCallDest` which does not exist in `main` yet) and mark it as ready for review. As far as I understand the semantics of "draft PR", the reviewers are only notified after the PR is marked as ready for review, thus after "undrafting" this PR, I would wait for at least another 24h in case there are any objections.

kbeyls reviewed (approved):
LGTM, thanks

atrosinenko commented:
Updated and un-drafted this PR after #131899 was merged into `main` branch. Resetting the approvals as the patch was modified significantly.

jacobbramley commented:
> @jacobbramley Do you suggest merging this first and then updating #131899?

I think that ordering makes sense, and is probably what I'd do, but I'm far from an expert in GitHub workflows (and I'm not a regular LLVM contributor so I don't know the local conventions well).

jacobbramley reviewed (approved):
LGTM, but I'm not sure if my approval is sufficient here.

atrosinenko commented:
>> @jacobbramley Do you suggest merging this first and then updating #131899?
> 
> I think that ordering makes sense, and is probably what I'd do, but I'm far from an expert in GitHub workflows (and I'm not a regular LLVM contributor so I don't know the local conventions well).

Probably some outdated notification was sent only after the PR was marked as ready for review. Actually, #131899 is already merged - that is the reason for updating this PR.

kbeyls reviewed (approved):
LGTM!
